File: 13Programmers/kakao_2020_str_press.py
def solution(s):
    answer = len(s)
    
    for i in range( 1, len(s)//2 + 1):
        result = ""
        tmp = s[0:i]
        print("tmp",tmp)
        count = 1
        for j in range(i,len(s) + 1,i):
            if(tmp == s[j:j+i]):
                count +=1
                print("ASDASD")

            else:
                if(count == 1):
                    result += tmp
                else:
                    result= result + str(count) + tmp
                    

                tmp = s[j:j+i]
                count = 1
            print("result",result)    
        result += s[(len(s)//i)*i:]
        #print("s[(len(s)//i)*i:]",s[(len(s)//i)*i:])
        print("---------result",result)
        answer = min(answer, len(result))
    return answer



print(solution("abcabcabcabcdededededede"))

File: 13Programmers/kakao_2020_str_press.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(string s){
	int leng = s.size();
	int answer = leng;
	
	for ( int i = 1; i <=(leng/2); i++){
		string result = "";
		string tmp = s.substr(0,i);
		int count = 1;
		
		for(int j = i ; j <=leng ;j +=i){
			if(tmp == s.substr(j,i)){
				count ++;
			}
			else{
				if(count == 1){
					result += tmp;
				}
				else{
					result +=(to_string(count)+tmp);
				}
				
				tmp = s.substr(j,i);
				count = 1;
			}
		}
		result += s.substr((leng/i)*i);
		answer = min(answer,(int)result.length());
		
	}
	return answer;
}

